### ISSUE TITLE
Tensorflow Estimator API doesn't work in distributed mode

### ISSUE THREAD

**Issue by gravitywp** [tensorflow/tensorflow#19501]
here is my test code 

```
from tensorflow.python.keras.layers import Conv1D, MaxPooling1D
from tensorflow.python.keras.models import Model
import logging
level = logging.getLevelName('INFO')
logging.getLogger().setLevel(level)
model = tf.keras.Sequential()
output = Dense(2, activation="softmax")
model.add(Dense(64, activation="relu", input_shape=(10,)))
model.add(output)
model.compile('rmsprop', 'categorical_crossentropy')
est_model = tf.keras.estimator.model_to_estimator(keras_model=model)
train_input_fn = tf.estimator.inputs.numpy_input_fn(
        x={"dense_2_input": np.random.randint(10, size=(320, 10))},
        y=np.random.rand(320, 2),
        num_epochs=10000,
        shuffle=False)
est_model.train(train_input_fn)
```

My TF_CONFIG is like is
```
TF_CONFIG={
"cluster": {"chief": ["localhost:2223"], 
"worker": ["localhost:2221"], 
"ps": ["lcoalhost:2222"]}, 
"task": {"index": "0", "type": "chief"}
}
```

The chief is stuck on logging 
`INFO:tensorflow:Restoring parameters from /tmp/tmpe_c82nqn/keras_model.ckpt` 
and no ports is listening.
Maybe I can't simulate cluster in local? 

When I using strace to found out why? I saw a lots 
`[pid 16719] futex(0x7f4c39cab604, FUTEX_WAIT_BITSET_PRIVATE, 850, {4401792, 658466075}, ffffffff) = -1 ETIMEDOUT (Connection timed out)`

Without TF_CONFIG set, the code is working fine.

Any suggestion?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by drpngx** [comment]
The checkpoints need to be on a distributed filesystem, like NFS. Are you doing that?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by anj-s** [comment]
If you want to run your model across multiple machines you need to use Estimator's train_and_evaluate API.  train_and_evaluate uses a between graph async parameter server approach.

If you want to train your model across multiple GPUs you can use the new [DistributionStrategy APIs](https://www.tensorflow.org/versions/master/api_docs/python/tf/contrib/distribute/DistributionStrategy). Use model_to_estimator to convert your keras model to an estimator instance and use train as you have above.

**Comment by gravitywp** [comment]
Thanks, I'll look into it later.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Nilesh-Agrawal** [comment]
@anj-s  clarfications needed. Please help.
If I created a model using estimator then.

1. If i use distribution startegy as CollectiveAllReduceStrategy or ParameterServerStrategy through RunConfig then and estimator can be trained distributedly using estimator.train(), there is no need to use train_and_evaluate().
2. When specifying model_dir in RunConfig using parameter-server appraoch, the path to directory can be local machine path, it need not to be a shared file-system path.
3. When using ParameterServerStrategy, how to use synchronous and stale-synchronus approach, are SyncReplicasOptimizer and DropStaleGradientOptimizer compatible with Distribution strategy ParameterServerStrategy API.
